File: Resueltos_por_competidor/Gabriella_Castro/Minority.java
/*
 * Autor: Gabriella Castro
 * Problema: Minority
 * Juez online: VJudge
 * Veredicto: Accepted
 * URL: https://vjudge.net/problem/CodeForces-1633B
 */

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main{

	public static void main(String[] args) throws IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int t = Integer.parseInt(br.readLine());

		while (t-- > 0) {
			String s = br.readLine();
			int n = s.length();
			int ceros = 0;
			int unos = 0;

			for (char c : s.toCharArray()) {
				if (c == '0')
					ceros++;
				else
					unos++;
			}

			if (ceros != unos) {
				System.out.println(Math.min(ceros, unos));
			} else {
				System.out.println(ceros - 1);
			}
		}
	}

}

File: Resueltos_por_competidor/Miguel_Lopez/minority.cpp
/*
 * Autor: Miguel Lopez
 * Problema: Minority
 * Juez online: Codeforces
 * Veredicto: Accepted
 * Url: https://codeforces.com/problemset/problem/1633/B
 */

#include <bits/stdc++.h>
#include <cctype>
#include <cstdio>
#include <ios>
#include <ostream>
typedef long long int ll;
using namespace std;
void init_code() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    #ifndef ONLINE_JUDGE
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
    #endif 
}
int main() {
    init_code();
    int t; cin >> t;
    cin.ignore();
    while(t--) {
       // int mini = 1e9;
        string s; cin >> s;
        int conta = 0, contb = 0;
        for(char c: s) {
            if(c == '0') {
            conta++;
            } 
            else  {
            contb++;
            }
        }
        if(conta != contb) {
            cout << min(conta,contb) << endl;
        }
        else {
            cout << contb-1 << endl;
        }
    }
}

